clang/lib/3C/ProgramInfo.cpp

```diff
@@ -450,8 +450,12 @@ bool ProgramInfo::insertIntoExternalFunctionMap(ExternalFunctionMapType &Map,
           DiagnosticsEngine::Fatal, "duplicate definition for function %0");
       DE.Report(FD->getLocation(), DuplicateDefinitionsID).AddString(FuncName);
       exit(1);
+    } else {
+      // The old constraint has a body, but we've encountered another prototype
+      // for the function.
+      assert(OldC->hasBody() && !NewC->hasBody());
+      NewC->brainTransplant(OldC, *this);
     }
-    // else oldc->hasBody() so we can ignore the newC prototype
   }
   return RetVal;
```

Comment:
Another issue that predates the PR: It looks like this return value is supposed to tell the caller whether the current `FVConstraint` got inserted into the map or a previous one is being used. But looking through the call hierarchy in CLion, I can't find any code that actually uses the return value, so if it were incorrect, we wouldn't notice.  Should we change the function to return void?